### DISCUSSION TITLE
Opening behavior on macOS

### DISCUSSION THREAD

**Discussion by ThijmenDam** [filips123/PWAsForFirefox#287] [General]
Hi! I wonder if the current opening behavior of an app on macOS is intended. Right now, if I open a PWA that is already running (i.e. by double clicking the app icon or through Spotlight), a second instance of the app is opened. However, to me it would make more sense if the instance of the app that's already running would be brought to the front instead of opening a new instance. I think this is more in line with the usual behavior of macOS apps.

**Comment by filips123** [comment]
In 2.5.0, you will be able to set "When launching a web app that is already opened" to "Focus the existing window" to get this behavior.

**Reply by ThijmenDam** [reply]
Wonderful, thank you!

**Reply by asgeirtj** [reply]
Is this a macos related issue? Is there no workaround? Seems quite hard to use like this.

**Reply by filips123** [reply]
What? This behaviour can be configured in settings... By default it opens a new window as it's the same on all OSes, but if you want macOS-like behaviour, you can enable it.

**Reply by asgeirtj** [reply]
Can you show me where I configure this please?

**Reply by filips123** [reply]
You need to open the PWA, then in the PWA window, open settings (with the same menu in top right as you would in normal Firefox). You should see "Progressive Web Apps" section with option "When launching a web app that is already opened", and you should set that option to "Focus the existing window".

Or alternatively, open the PWA, press `F6` and enter `about:config`. On the config page, search for `firefoxpwa.launchType` and set it to `3`.

**Reply by asgeirtj** [reply]
Thank you for confirming the functionality. I would appreciate your advice regarding the efficacy of this extension compared to Progressive Web Apps (PWAs) on Chromium-based browsers. Is this approach inherently less robust? I have observed some performance issues, including slow window refocusing, frequent need for force quitting, and inconsistent app icon loading.

My primary motivation for exploring this option stems from my use of Firefox as my main browser and Chrome for PWAs. This setup presents two significant challenges: Chrome's tendency to consume excessive CPU and RAM resources, and the inconvenience of links from PWAs (e.g., Gmail, ChatGPT, Perplexity.ai) opening in Chrome rather than Firefox, despite my preference for the latter.

I have considered alternative solutions:

1. Utilizing the Velja app, which allows for right-click "Open with Velja" functionality, configurable to use Firefox. However, this process is somewhat cumbersome. I have contemplated creating a BetterTouchTool action to streamline this with a modifier+left click combination.

2. Implementing the "Open in Firefox" Chrome extension. Despite attempts to configure it in reverse mode and whitelist specific domains to open in Firefox, I have not achieved consistent results.

My preference for Firefox is partly due to its superior performance on macOS. Therefore, the ability to create and utilize PWAs within Firefox would be an ideal solution. However, the current implementation challenges are concerning. The extended time required to refocus PWAs (approximately 3 seconds) and issues with external links in certain applications (e.g., Gmail) are notable drawbacks. Additionally, the need to adjust settings for specific scenarios, such as Google Authentication in ChatGPT, adds complexity.

In light of these observations, I am seeking clarification: Are these issues indicative of my inexperience with the extension, or are they inherent limitations? Is it feasible to achieve my desired functionality through proper configuration of this extension?

Thank you for your insights.
